compiler/luci/import/src/ImportMetadata.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ImportMetadata.h"
+
+#include <vector>
+
+namespace
+{
+
+/**
+ * 'op_table'
+ * - [ id : uint32_t ][ node_num : uint32_t ][ node_ids : node_num * uint32_t ]
+ * - 'node_num' should be larger than zero. (TODO Check it!)
+ */
+std::map<uint32_t, std::set<uint32_t>> metadata_with_op_table(const std::vector<uint8_t> &buffer)
+{
+  std::map<uint32_t, std::set<uint32_t>> node_source_ids_map;
+
+  uint32_t idx = 0;
+  while (idx < buffer.size())
+  {
+    uint32_t id = 0;
+    id += (buffer.at(idx + 0) << 0 * 8);
+    id += (buffer.at(idx + 1) << 1 * 8);
+    id += (buffer.at(idx + 2) << 2 * 8);
+    id += (buffer.at(idx + 3) << 3 * 8);
+    idx += 4;
+
+    uint32_t node_num = 0;
+    node_num += (buffer.at(idx + 0) << 0 * 8);
+    node_num += (buffer.at(idx + 1) << 1 * 8);
+    node_num += (buffer.at(idx + 2) << 2 * 8);
+    node_num += (buffer.at(idx + 3) << 3 * 8);
+    idx += 4;
+
+    std::set<uint32_t> source_ids;
+    for (uint32_t j = 0; j < node_num; ++j)
+    {
+      uint32_t origin = 0;
+      origin += (buffer.at(idx + 4 * j + 0) << 0 * 8);
+      origin += (buffer.at(idx + 4 * j + 1) << 1 * 8);
+      origin += (buffer.at(idx + 4 * j + 2) << 2 * 8);
+      origin += (buffer.at(idx + 4 * j + 3) << 3 * 8);
+
+      source_ids.insert(origin);
+    }
+    idx += (node_num * 4);
+
+    if (node_source_ids_map.insert({id, source_ids}).second == false)
+      throw std::runtime_error("Duplicated origin ID");
+  }
+  assert(idx == buffer.size());
+
+  return node_source_ids_map;
+}
+
+/**
+ * 'source_table'
+ * - [ id : uint32_t ][ length : uint32_t ][ data : 'length' Bytes ]
+ */
+std::map<uint32_t, std::string> metadata_with_source_table(const std::vector<uint8_t> &buffer)
```

Comment:
(Originally written by @seanshpark)

how about declare a type (struct or class) for `std::map<uint32_t, std::string>` ?

Reply:
As this type is used only in `CircleMetadataImporter.cpp`, leaving as it is seems not bad for me :D